### PR TITLE
Upgrade netty from 4.1.73.Final to 4.1.78.Final

### PR DESCRIPTION
**Pull request by cwperks**
Signed-off-by: Craig Perkins <cwperx@amazon.com>

### Description

First PR! 🥇 This upgrades Netty to address a CVE in 4.1.73. This resolves [1831](https://github.com/opensearch-project/security/issues/1831)
 
### Issues Resolved

- [1831](https://github.com/opensearch-project/security/issues/1831)
 
### Check List

- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by reta** [comment]
I am curios why @dependabot didn't do that ...

File: modules/transport-netty4/src/javaRestTest/java/org/opensearch/rest/Netty4BadRequestIT.java [hunk 1]
```diff
@@ -86,7 +86,7 @@ public void testBadRequest() throws IOException {
             () -> client().performRequest(new Request(randomFrom("GET", "POST", "PUT"), path))
         );
         assertThat(e.getResponse().getStatusLine().getStatusCode(), equalTo(BAD_REQUEST.getStatus()));
-        assertThat(e, hasToString(containsString("too_long_frame_exception")));
+        assertThat(e, hasToString(containsString("too_long_http_line_exception")));
```
Comment (cwperks) [inline modules/transport-netty4/src/javaRestTest/java/org/opensearch/rest/Netty4BadRequestIT.java:89]: This message changed from `4.1.74.Final -> 4.1.75.Final`.

Details here: https://github.com/netty/netty/commit/3ba2eed16a627371839c63113cdb39df9d6bfe35#diff-864e434ddf7f115156d8497898df1bb48240c9d488d6652b52c437bf9c91fb96

**Comment by saratvemulapalli** [comment]
> I am curios why @dependabot didn't do that ...

hm.. may be @VachaShah might know about it.

**Review by saratvemulapalli — APPROVED** [review]

**Comment by VachaShah** [comment]
> > I am curios why @dependabot didn't do that ...
> 
> hm.. may be @VachaShah might know about it.

I think Dependabot is not able to scan `version.properties` file.

**Comment by saratvemulapalli** [comment]
@VachaShah @reta I've opened up an issue https://github.com/opensearch-project/OpenSearch/issues/3782. 
I have no idea if it can be done, feel free to chime if you have ideas to make it happen.

**Pull request merged**